diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/SAPDB/WebAgent/HTTP/WAHTTP_DefaultDocument.html V74_03_17.chng/SAPDB_ORG/sys/src/SAPDB/WebAgent/HTTP/WAHTTP_DefaultDocument.html
--- V74_03_17.orig/SAPDB_ORG/sys/src/SAPDB/WebAgent/HTTP/WAHTTP_DefaultDocument.html	Thu Mar 20 15:35:53 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/SAPDB/WebAgent/HTTP/WAHTTP_DefaultDocument.html	Sat May  3 06:02:25 2003
@@ -27,8 +27,17 @@
 </head>
 
 <body background="/WARoot/Images/tatami.gif" topmargin="0" leftmargin="0">
-Welcome to the world of SAP DB</br></br>
-Your SAP DB Web Server is now running...
+<h1 align="center">Welcome to the world of SAP DB on 
+<font color="green">PLD Linux</font></h1>
+<center><b>Your SAP DB Web Server is now running...</b></center></br>
+
+<i><b>Available tools:</i></b></br>
+<ul>
+<li><a href="/websql">WEB SQL</a>
+<li><a href="/webdbm">WEB Database Manager</a>
+<li><a href="/waadmin.wa">WEB Admin</a>
+<li><a href="/waecho">Echo</a>
+</ul>
 </body>
 
 </html>
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMBackup.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMBackup.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMBackup.htm.6w	Thu Mar 20 15:38:29 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMBackup.htm.6w	Sat May  3 06:10:44 2003
@@ -139,7 +139,7 @@
     back = "Event=BACKUP&Action=REFRESH";
     back = back + "&BackupType=" + backuptype;
     back = back + "&MediumType=" + mediumtype;
-    s = s + "&Back=\"" + parent.CreateWebDBMURL(back) + "\"";
+    s = s + "&Back=\"" + escape(parent.CreateWebDBMURL(back)) + "\"";
 
     this.location.href = parent.CreateWebDBMURL(s);
   }
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMCommand.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMCommand.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMCommand.htm.6w	Thu Mar 20 15:38:30 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMCommand.htm.6w	Sat May  3 06:11:54 2003
@@ -143,4 +143,4 @@
 
   </body>
 
-</html>
\ No newline at end of file
+</html>
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMDevspaces.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMDevspaces.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMDevspaces.htm.6w	Thu Mar 20 15:38:30 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMDevspaces.htm.6w	Sat May  3 06:13:15 2003
@@ -16,17 +16,17 @@
   }
   function EditDevspace(name) {
     var s;
-    s = "Event=CONFIG_DEVSPACE&Action=EDIT&Name=" + name + "&Back=\"" + this.location.href + "\"";
+    s = "Event=CONFIG_DEVSPACE&Action=EDIT&Name=" + name + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function AddDevspace(classname) {
     var s;
-    s = "Event=CONFIG_DEVSPACES&Action=ADD&Class=" + classname + "&Back=\"" + this.location.href + "\"";
+    s = "Event=CONFIG_DEVSPACES&Action=ADD&Class=" + classname + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function EditParam(name) {
     var s;
-    s = "Event=CONFIG_PARAM&Action=EDIT&Name=" + name + "&Back=\"" + this.location.href + "\"";
+    s = "Event=CONFIG_PARAM&Action=EDIT&Name=" + name + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function ShowSize(pages) {
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMFiles.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMFiles.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMFiles.htm.6w	Thu Mar 20 15:38:30 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMFiles.htm.6w	Sat May  3 06:13:30 2003
@@ -387,4 +387,4 @@
 
 </body>
 
-</html>
\ No newline at end of file
+</html>
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMMedia.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMMedia.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMMedia.htm.6w	Thu Mar 20 15:38:30 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMMedia.htm.6w	Sat May  3 06:22:33 2003
@@ -24,14 +24,14 @@
     s = "Event=CONFIG_MEDIUM&Action=EDIT";
     s = s + "&Type=" + type;
     s = s + "&Name=" + name;
-    s = s + "&Back=\"" + this.location.href + "\"";
+    s = s + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function AddMedium(type) {
     var s;
     s = "Event=CONFIG_MEDIA&Action=ADD";
     s = s + "&Type=" + type;
-    s = s + "&Back=\"" + this.location.href + "\"";
+    s = s + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function ShowSize(pages) {
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMParams.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMParams.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMParams.htm.6w	Thu Mar 20 15:38:31 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMParams.htm.6w	Sat May  3 06:17:59 2003
@@ -16,7 +16,7 @@
   }
   function EditParam(name) {
     var s;
-    s = "Event=CONFIG_PARAM&Action=EDIT&Name=" + name + "&Back=\"" + this.location.href + "\"";
+    s = "Event=CONFIG_PARAM&Action=EDIT&Name=" + name + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function ParamName(name,change) {
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMRecovery.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMRecovery.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMRecovery.htm.6w	Thu Mar 20 15:38:31 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMRecovery.htm.6w	Sat May  3 06:18:31 2003
@@ -28,7 +28,7 @@
       var s;
       s = "Event=CONFIG_MEDIA&Action=ADD";
       s = s + "&Type=" + mediumtype;
-      s = s + "&Back=\"" + this.location.href + "\"";
+      s = s + "&Back=\"" + escape(this.location.href) + "\"";
       this.location.href = parent.CreateWebDBMURL(s);
     }
 
@@ -37,7 +37,7 @@
       s = "Event=CONFIG_MEDIUM&Action=EDIT";
       s = s + "&Type=" + mediumtype;
       s = s + "&Name=" + name;
-      s = s + "&Back=\"" + this.location.href + "\"";
+      s = s + "&Back=\"" + escape(this.location.href) + "\"";
       this.location.href = parent.CreateWebDBMURL(s);
     }
 
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMUsers.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMUsers.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMUsers.htm.6w	Thu Mar 20 15:38:31 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMUsers.htm.6w	Sat May  3 06:20:30 2003
@@ -17,13 +17,13 @@
     var s;
     s = "Event=CONFIG_USER&Action=EDIT";
     s = s + "&Name=" + name;
-    s = s + "&Back=\"" + this.location.href + "\"";
+    s = s + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function Add() {
     var s;
     s = "Event=CONFIG_USERS&Action=ADD";
-    s = s + "&Back=\"" + this.location.href + "\"";
+    s = s + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
 </script>
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMWizard.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMWizard.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/DBMWizard.htm.6w	Thu Mar 20 15:38:31 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/DBMWizard.htm.6w	Sat May  3 06:24:53 2003
@@ -70,14 +70,14 @@
     s = "Event=CONFIG_MEDIUM&Action=EDIT";
     s = s + "&Type=" + type;
     s = s + "&Name=" + name;
-    s = s + "&Back=\"" + this.location.href + "\"";
+    s = s + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function AddMedium(type) {
     var s;
     s = "Event=CONFIG_MEDIA&Action=ADD";
     s = s + "&Type=" + type;
-    s = s + "&Back=\"" + this.location.href + "\"";
+    s = s + "&Back=\"" + escape(this.location.href) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function ShowParams(group) {
@@ -92,7 +92,7 @@
     var s;
     var backurl;
     backurl = parent.CreateWebDBMURL("Event=WIZARD&Action=VIEW&Step=" + step + "&Group=" + paramgroup);
-    s = "Event=CONFIG_PARAM&Action=EDIT&Name=" + name + "&Back=\"" + backurl + "\"";
+    s = "Event=CONFIG_PARAM&Action=EDIT&Name=" + name + "&Back=\"" + escape(backurl) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function ParamName(name,change) {
@@ -116,7 +116,7 @@
     var s;
     var backurl;
     backurl = parent.CreateWebDBMURL("Event=WIZARD&Action=VIEW&Step=" + step + "&Class=" + devspaceclass);
-    s = "Event=CONFIG_DEVSPACE&Action=INIT&Name=" + name + "&Back=\"" + backurl + "\"";
+    s = "Event=CONFIG_DEVSPACE&Action=INIT&Name=" + name + "&Back=\"" + escape(backurl) + "\"";
     this.location.href = parent.CreateWebDBMURL(s);
   }
   function Install() {
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/SAPDBMsgBox.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/SAPDBMsgBox.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/SAPDBMsgBox.htm.6w	Thu Mar 20 16:38:32 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/SAPDBMsgBox.htm.6w	Mon May 19 13:57:15 2003
@@ -7,7 +7,7 @@
     <title>SAP DB Web Access</title>
   </head>
 
-  <body topmargin="0" leftmargin="0" marginwidth="0" marginheight="0" background="/WARoot/images/tatami.gif">
+  <body topmargin="0" leftmargin="0" marginwidth="0" marginheight="0" background="/WARoot/Images/tatami.gif">
 
     <!-- TemplateKey Logo -->
     <table border="0" cellspacing="0" cellpadding="0" width=100%>
diff -Nur V74_03_17.orig/SAPDB_ORG/sys/src/wd/SAPDBSimpleTable.htm.6w V74_03_17.chng/SAPDB_ORG/sys/src/wd/SAPDBSimpleTable.htm.6w
--- V74_03_17.orig/SAPDB_ORG/sys/src/wd/SAPDBSimpleTable.htm.6w	Thu Mar 20 16:38:32 2003
+++ V74_03_17.chng/SAPDB_ORG/sys/src/wd/SAPDBSimpleTable.htm.6w	Mon May 19 13:57:20 2003
@@ -6,7 +6,7 @@
     <script language="JavaScript" src="/WARoot/JavaScripts/wqbutton.js"></script>
   </head>
 
-  <body topmargin="0" leftmargin="0" marginwidth="0" marginheight="0" background="/WARoot/images/tatami.gif">
+  <body topmargin="0" leftmargin="0" marginwidth="0" marginheight="0" background="/WARoot/Images/tatami.gif">
 
     <!-- TemplateKey Logo -->
     <table border="0" cellspacing="0" cellpadding="0" width=100%>
